geps/gep-3779/index.md

```diff
@@ -42,20 +37,427 @@ Kubernetes core provides NetworkPolicies as one way to do it. Network Policies h
+    //
+    // Note for reviewers: While this GEP primarily focuses on identity-based
+    // authorization where identity is often established at the transport layer,
+    // some implementations might derive identity from authenticated tokens or sources
+    // within the request itself.
+    //
+    // +optional
+    SPIFFE *AuthorizationSourceSPIFFE `json:"spiffe,omitempty"`
+
+    // ServiceAccount specifies a Kubernetes Service Account that is
+    // matched by this rule. A request originating from a pod associated with
+    // this serviceaccount will match the rule.
+    //
+    // The ServiceAccount listed here is expected to exist within the same
+    // trust domain as the targeted workload. Cross-trust-domain access
+    // should instead be expressed using the `SPIFFE` field.
+    // +optional
+    ServiceAccount AuthorizationSourceServiceAccount `json:"serviceAccount,omitempty"`
+}
+
+// +kubebuilder:validation:XValidation:rule="self.startsWith("spiffe://") && size(self.split("/")) >= 3", message="spiffe source must start with spiffe:// and include at least / separating trust domain from the workload identity."
+type AuthorizationSourceSPIFFE AbsoluteURI
+
+type AuthorizationSourceServiceAccount struct {
+  // Namespace is the namespace of the ServiceAccount
+  // If not specified, current namespace (the namespace of the policy) is used.
+  Namespace *Namespace `json:"namespace,omitempty"`
+
+  // Name is the name of the ServiceAccount. 
+  // Use "*" to indicate all serviceaccounts in the namespace.
+  // +required
+  Name string `json:"name"`
+}
+
+
+// AuthorizationAttribute defines L4 properties of a request destination.
+type AuthorizationNetworkAttributes struct {
+    // Ports specifies a list of destination ports to match on.
+    // Traffic is matched if it is going to any of these ports.
+    // If not specified, the rule applies to all ports.
+    // +optional
+    Ports []gatewayv1.PortNumber `json:"ports,omitempty"`
+}
+
+```
+
+Note: More advanced logic like negation or explicit Boolean operations is left for potential future work.
+
+## Future Enhancements
+
+### DENY Policies
+
+We start without DENY policies in scope, DENY policies _may_ be added in future iterations.
+
+### Future L7 Support
+
+It is very likely that graduation of this API will also require adding L7 support to it. However, after numerous conversations on this topic, the decision is to start with L4 only and leave L7 for future iterations. The details of how L7 support could look like is not in scope for this GEP.
 
+## Graduation Criteria and Guardrails
+
+This GEP is starting as Experimental with X-prefix. We would like to define some guardrails to avoid being stuck with experimental api in the project forever.
+
+GEP is removed after 3 releases unless:
+
+  1. There is more than 1 implementation (implementations that are nearing completion can count here)
+  2. There are conformance tests in place
+
```

Comment:
```suggestion
3. The label selector pattern is added to the policy attachment definition in GEP-713.

```

From discussion in https://github.com/kubernetes-sigs/gateway-api/pull/3891/files#r2551591308, we don't want to encourage this pattern elsewhere or add it to GEP-713 until we determine it is necessary, but we shouldn't graduate this resource to standard while this pattern is still experimental.

@guicassolato @LiorLieberman @kflynn